internal/provider/inventory_data_source.go

```diff
@@ -66,12 +72,24 @@ func (d *InventoryDataSource) Read(ctx context.Context, req datasource.ReadReque
 		return
 	}
 
+	newUUID, err := uuid.NewUUID()
+	if err != nil {
+		resp.Diagnostics.AddError("failed to generate unique ID for inventory", err.Error())
+		return
+	}
+	data.Id = types.StringValue(newUUID.String()[:8])
+
 	f, err := os.CreateTemp("", "imagetest-")
 	if err != nil {
 		resp.Diagnostics.AddError("failed to create temp file", err.Error())
 		return
 	}
-	defer f.Close()
+	defer func() {
+		closeErr := f.Close()
+		if closeErr != nil {
+			panic(fmt.Errorf("failed to close temporary file: %w", closeErr))
```

Comment:
imo we just log/warn on this, panic feels aggressive for `/tmp` closes

Reply:
Updated: 781e2e2d73c523baedf675cfda3f8c93f1899c1f